src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1650,40 +1650,44 @@ BOOST_AUTO_TEST_CASE(GradientConcatenateLayerTest)
  */
 BOOST_AUTO_TEST_CASE(SimpleLookupLayerTest)
 {
-  arma::mat output, input, delta, gradient;
-  Lookup<> module(10, 5);
+  const size_t vocabSize = 10;
+  const size_t embeddingSize = 2;
+  const size_t seqLength = 3;
+  const size_t batchSize = 4;
+
+  arma::mat output, input, gy, g, gradient;
+
+  Lookup<> module(vocabSize, embeddingSize);
   module.Parameters().randu();
 
   // Test the Forward function.
-  input = arma::zeros(2, 1);
-  input(0) = 1;
-  input(1) = 3;
+  input = arma::zeros(seqLength, batchSize);
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    int token = math::RandInt(1, vocabSize);
+    input(i) = token;
+  }
 
   module.Forward(input, output);
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    // The Lookup module uses index - 1 for the cols.
+    const double outputSum = arma::accu(module.Parameters().cols(
+      arma::conv_to<arma::uvec>::from(input.col(i)) - 1));
 
-  // The Lookup module uses index - 1 for the cols.
-  const double outputSum = arma::accu(module.Parameters().col(0)) +
-      arma::accu(module.Parameters().col(2));
-
-  BOOST_REQUIRE_CLOSE(outputSum, arma::accu(output), 1e-3);
+    BOOST_REQUIRE_CLOSE(outputSum, arma::accu(output.col(i)), 1e-3);
+  }
 
   // Test the Backward function.
-  module.Backward(input, input, delta);
-  BOOST_REQUIRE_EQUAL(arma::accu(input), arma::accu(input));
+  gy = 0.3 * arma::randu(embeddingSize * seqLength, batchSize);
+  module.Backward(input, gy, g);
+  BOOST_REQUIRE_EQUAL(arma::accu(gy), arma::accu(g));
 
```

Comment:
To me, the previous test didn't make any sense. It was not a concern because `Lookup` is generally used as the first layer, so only gradient update is required. But still, I thought to correct that.

Reply:
Yeah, make sense.